src/libraries/Common/src/System/IO/Hashing/NonCryptographicHashAlgorithm.cs

```diff
@@ -14,7 +14,13 @@ namespace System.IO.Hashing
     /// <summary>
     ///   Represents a non-cryptographic hash algorithm.
     /// </summary>
-    public abstract class NonCryptographicHashAlgorithm
+#if SYSTEM_PRIVATE_CORELIB
+#pragma warning disable CA1512
+    internal
```

Comment:
Why do we need to bring in the abstraction? We're calling into the static entrypoints; can we not move some of the implementation around so that we only need to bring the static code paths into corelib?

Reply:
Addressed